### ISSUE TITLE
Mongo Panache error with '$in' queries on a CustomID field

### ISSUE THREAD

**Issue by pclainchard** [quarkusio/quarkus#19840]
### Describe the bug

While using CustomIds, a query $in doesn' work (and returns an empty array)

I pushed a sample project to reproduce this issue : https://github.com/pclainchard/quarkusPanacheIssueCustomID 

- I made two services, one that uses the CustomID field, and an other that uses a regular field and works
- I have tested it with both Long and String CustomIDs and the behavior is the same

I have discussed this issue with @loicmathieu 

### Expected behavior

A $in query based on a CustomID field should retrieve the documents with theses ids

For example : 

```java
  @Path("/findByField")
    public ArrayList<UserModel> listByOtherField(@QueryParam("id") ArrayList<String> listIds) {
        ArrayList<UserModel> result = new ArrayList<UserModel>();

        PanacheQuery<PanacheMongoEntityBase> query = UserDAO.find("{'customId' :{$in: [?1]}}", listIds);
```

### Actual behavior

An empty array is returned for a query based on a CustomID field, but works fine for an other field

### How to Reproduce?

1. Create a customID on a field by using @BsonId annotation on a class extending **PanacheMongoEntityBase**
2. Add function that uses this Class (for example UserDAO) and add a Query **find** (eg :  UserDAO.find("{'userId' :{$in: [?1]}}", listIds);

In my sample project, I've added a ".http" file that contains requests that can be executed https://github.com/pclainchard/quarkusPanacheIssueCustomID/blob/master/src/test/resources/sampleRequests.http 

IntelliJ allows you to execute .http files queries natively, in VSCode you have to use the RESTClient extension

### Output of `java -version`

openjdk version "16.0.2" 2021-07-20

### Quarkus version or git rev

2.2.1.final

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Apache Maven 3.8.1

**Comment by pclainchard** [comment]
I think it is related to the fact that a Custom ID is store "_id" in the Mongo database...

So my query cannot find any field that is named "CustomId"

I made a loop to bypass this behavior : loop on my IDs, and then for each, made a "findById" request

Maybe an improvement would be to add a "findByIds" method that could take a list of ObjectId values ?

So finally, I would consider this issue not to be an issue, and maybe I could write an adendum for the Documentation just to explain this ? :)

**Comment by loicmathieu** [comment]
> I think it is related to the fact that a Custom ID is store "_id" in the Mongo database...

Yes, so your query must find on `_id` and not `userId`, at least on the current version of MongoDB with Panache.

**Comment by loicmathieu** [comment]
Today you use:

```
UserDAO.find("{'userId' :{$in: [?1]}}", listIds);
```

And replacing userId by _id will do the trick.

```
UserDAO.find("{'userId' :{$in: [?1]}}", listIds);
// this one is with PanacheQL
UserDAO.find("_id in ?1", listIds);
```

Today, we allow using object field name instead of database field name in case of `@BsonProperty` but not when using `@BsonId` I think it's an oversight worth fixing it. This issue will occurs not only on $in but for all queries based on custom ID when using the object field name instead of _id.

**Comment by anamak9** [comment]
what about $nin operator ? can somebody address the below ?
https://stackoverflow.com/questions/76580062/quarkus-mongodb-panache-way-to-do-nin

**Comment by loicmathieu** [comment]
@anamak9 as I comment in the stackoverflow, please open a feature request in this repository to translate NOT IN to $nin.
